### PR TITLE
After partitioning we now retry the stat prior to formatting

### PR DESCRIPTION
**Pull request by dprotaso**
**- What I did**
After partitioning make waiting for the device durable again

**- How I did it**

stat can fail - simply don't exit when it does

**- How to verify it**

This is hard to reproduce but we were getting this bug with fresh disks numerous times today

**- Description for the changelog**
format pkg has retries `stat` again when waiting for the partition device

### CODE REVIEW

File: pkg/format/format.go [hunk 1]
```diff
@@ -100,21 +100,27 @@ func refreshDevicesAndWaitFor(awaitedDevice string) error {
...
 		time.Sleep(100 * time.Millisecond)
 		exec.Command("mdev", "-s").Run()
 	}
 	if !done {
-		return fmt.Errorf("Error waiting for device %s", awaitedDevice)
+		var statMsg string
+		if err != nil {
+			statMsg = fmt.Sprintf(" - stat returned: %v", err)
+		}
+		return fmt.Errorf("Error waiting for device %s%s", awaitedDevice, statMsg)
```
Comment (eyz) [inline pkg/format/format.go]: Maybe change to this instead? 
`return fmt.Errorf("Failed to find block device %s%s", awaitedDevice, statMsg)`

**Comment by justincormack** [comment]
@dave-tucker is in SF at the minute so might be a delay reviewing...

**Comment by justincormack** [comment]
Which error are you getting from `stat` when it fails?

**Comment by rn** [comment]
I would also be good to add a bit more detail to the commit message (instead of the PR). This way they are stored in git.

**Comment by dprotaso** [comment]
/dev/vda1 not found

**Comment by rn** [comment]
@dprotaso thanks for the update. The changes look good to me. Could you update all uses of the `format` package in the linuxkit repo *in a separate commit*. I've build/pushed/signed the udapte and the new tag is: `linuxkit/format:6b46d0450082f397177da36be6b4d74d93eacd1e`

There is a script which helps you updating the YAML files. From the top-level:
```
./scripts/update-component-sha.sh linuxkit/format:10e75e78e1f134d310c62e9a0352df1c67b0dd81 linuxkit/format:6b46d0450082f397177da36be6b4d74d93eacd1e
```
should do the right thing and allow CI to test your change.

**Comment by justincormack** [comment]
Will build the new package...

**Comment by rn** [comment]
@justincormack new package is already build/pushed. Just needs update of YAMLs

**Pull request merged**